src/Avalonia.Controls.TreeDataGrid/IndexPath.cs

```diff
@@ -145,17 +155,22 @@ static IEnumerator<int> EnumerateSingleOrEmpty(int index)
         }
 
         public override int GetHashCode()
+        {
+            return _hashCode == 0 ? CalculateHashCode(_path, _index) : _hashCode;
```

Comment:
Why do we need this check here?
```
return _hashCode == 0 ? CalculateHashCode(_path, _index) : _hashCode;
```

I would expect to "just" return `_hashCode` since it's calculated in the constructor.

Reply:
This is for when the `IndexPath` is created as the `default` object (e.g. `row.ModelIndex == default`). In this case the hash code should be the same as the one in an empty `IndexPath`, but it's 0 because the constructor is not called